### PR TITLE
feat: Add mentionRaidProtectionEnabled

### PR DESCRIPTION
**Pull request by Harbinger-of-Death**
- This adds a settable trigger metadata property called mentionRaidProtectionEnabled.
- Also adds separate methods for setting trigger metadatas

### CODE REVIEW

File: Structures/AutoModeration.js [hunk 1]
```diff
@@ -16,14 +17,33 @@ class AutoModeration extends Base {
...
         this.actions = data.actions?.map(o => { return { type: o.type, metadata: { channelId: o.metadata?.channel_id, durationSeconds: o.metadata?.duration_seconds } } }) ?? []
         this.createdAt = data.id ? Snowflake.deconstruct(data.id).createdAt : null
         this.createdTimestamp = this.createdAt?.getTime() ?? null
     }
 
+    get exemptChannels() {
+        const collection = new Collection()
+        for(const snowflake of this._data.exempt_channels) {
+            if(!this.client.channels.cache.has(snowflake)) continue;
+            collection.set(snowflake, this.client.channels.cache.get(snowflake))
```
Comment (MrMythicalYT) [inline Structures/AutoModeration.js]:
```suggestion
        for(const channel of this._data.exempt_channels) {
            if(!this.client.channels.cache.has(channel)) continue;
            collection.set(channel, this.client.channels.cache.get(channel))
```

File: Structures/AutoModeration.js [hunk 2]
```diff
@@ -16,14 +17,33 @@ class AutoModeration extends Base {
...
+        }
+
+        return collection
+    }
+
+    get exemptRoles() {
+        const collection = new Collection()
+        for(const snowflake of this._data.exempt_roles) {
+            if(!this.guild?.roles.cache.has(snowflake)) continue;
+            collection.set(snowflake, this.guild?.roles.cache.get(snowflake))
```
Comment (MrMythicalYT) [inline Structures/AutoModeration.js]:
```suggestion
        for(const role of this._data.exempt_roles) {
            if(!this.guild?.roles.cache.has(role)) continue;
            collection.set(role, this.guild?.roles.cache.get(role))
```

**Review by MrMythicalYT — CHANGES_REQUESTED** [review]
More descriptive variable names

File: index.d.ts [hunk 3]
```diff
@@ -654,6 +654,10 @@ export class AutoModeration extends Base {
      * Total number of unique Roles and User mentions allowed per Message. For MentionSpam trigger type
      */
     public mentionTotalLimit: number
+    /**
+     * Whether or not this Automod has mention protection enabled against raid. For MentionSpam trigger type
+     */
+    public mentionRaidProtectionEnabled: boolean
```
Comment (Harbinger-of-Death) [inline index.d.ts:660]: New undocumented property

Comment (Harbinger-of-Death) [reply index.d.ts:660]: don't tell discord

**Pull request merged**